libs/hdf-converters/src/ckl-mapper/checklist-mapper.ts

```diff
@@ -0,0 +1,354 @@
+   * Constructor takes an addtional (optional) param object supplementalInfo
+   * @param checklistXml - string of xml data
+   */
+  constructor(checklistXml: string) {
+    super();
+    this.checklistXml = checklistXml;
+    this.raw = super.toJsonix(
+      this.checklistXml,
+      checklistMapping.jsonixMapping
+    );
+    this.checklistObject = super.toIntermediateObject(this.raw);
+  }
+
+  toHdf(): ExecJSON.Execution {
+    const numberOfStigs = this.checklistObject.stigs.length;
+    if (numberOfStigs === 1) {
+      const defaultChecklist = new ChecklistMapper(this.checklistObject);
+      return defaultChecklist.toHdf();
+    } else {
+      const checklist = new ChecklistMapper(this.checklistObject);
+      const original = checklist.toHdf();
+      const parentProfileName = 'Parent Profile';
+      const parent_profile: ExecJSON.Profile = {
+        name: parentProfileName,
+        version: HeimdallToolsVersion,
+        supports: [],
+        attributes: [],
+        groups: [],
+        depends: [],
+        controls: [],
+        sha256: ''
+      };
+      for (const profile of original.profiles) {
+        parent_profile.depends?.push({name: profile.name});
+        parent_profile.controls.push(...profile.controls);
+        profile.parent_profile = parentProfileName;
+        profile.sha256 = generateHash(JSON.stringify(profile));
+      }
+      parent_profile.sha256 = generateHash(JSON.stringify(parent_profile));
+      original.profiles.push(parent_profile);
+      console.log('Bottom of HDF Converter');
+      return original;
+    }
+  }
+}
+
+/**
+ * Checklist mapper
+ */
+export class ChecklistMapper extends BaseConverter {
+  withRaw: boolean;
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {},
+    profiles: [
+      {
+        path: 'stigs',
+        name: {path: 'header.title'},
+        version: {path: 'header.version'},
+        title: {path: 'header.title'},
+        summary: {path: 'header.description'},
+        license: {path: 'header.notice'},
+        supports: [],
+        attributes: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'vulns',
+            key: 'id',
+            tags: {
+              gtitle: {path: 'groupTitle'},
+              rid: {path: 'ruleId'},
+              gid: {path: 'vulnNum'},
+              stig_id: {path: 'ruleVersion'},
+              cci: {
+                path: 'cciRef',
+                transformer: cciRef
+              },
+              nist: {
+                path: 'cciRef',
+                transformer: nistTag
+              },
+              weight: {path: 'weight'},
+              transformer: (input: ChecklistVuln): Record<string, unknown> => {
```

Comment:
probably a good idea to leave a comment explaining the internal data representation vs label in hdf tags thing